src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -507,16 +538,30 @@ void DxFontRenderData::_SetFeatures(const std::unordered_map<std::wstring_view,
 void DxFontRenderData::_SetAxes(const std::unordered_map<std::wstring_view, float>& axes)
 {
     _axesVector.clear();
+    _didUserSetAxes = false;
+    _didUserSetItalic = false;
 
     // Update our axis map with the provided axes
-#pragma warning(suppress : 26445) // the analyzer doesn't like reference to string_view
-    for (const auto& [axis, value] : axes)
+    if (!axes.empty())
     {
-        if (axis.length() == TAG_LENGTH)
+#pragma warning(suppress : 26445) // the analyzer doesn't like reference to string_view
+        for (const auto& [axis, value] : axes)
         {
-            const auto dwriteTag = DWRITE_MAKE_FONT_AXIS_TAG(til::at(axis, 0), til::at(axis, 1), til::at(axis, 2), til::at(axis, 3));
-            _axesVector.emplace_back(DWRITE_FONT_AXIS_VALUE{ dwriteTag, value });
+            if (axis.length() == TAG_LENGTH)
+            {
+                const auto dwriteFontAxis = DWRITE_FONT_AXIS_VALUE{ DWRITE_MAKE_FONT_AXIS_TAG(til::at(axis, 0), til::at(axis, 1), til::at(axis, 2), til::at(axis, 3)), value };
+                _axesVector.emplace_back(dwriteFontAxis);
+                if (dwriteFontAxis.axisTag != DWRITE_FONT_AXIS_TAG_WEIGHT)
+                {
+                    _axesVectorWithoutWeight.emplace_back(dwriteFontAxis);
+                }
+                if (dwriteFontAxis.axisTag == DWRITE_FONT_AXIS_TAG_ITALIC && value == 1)
+                {
+                    _didUserSetItalic = true;
+                }

```

Comment:
Personally I'd shorten these things to
```cpp
_didUserSetItalic |= dwriteFontAxis.axisTag == DWRITE_FONT_AXIS_TAG_ITALIC && value == 1;
```

The benefits as I personally, subjectivelly perceive them:
* Easier to read.
* It compiles to what GCC/Clang would produce. MSVC always needs a bit of handholding when it comes to booleans.
* And a longer one...

CPUs appear to be really fast nowadays, but they're actually often just as dumb as they used to be. One of the problems is that the "predictor" for branches has a limited size. While an important branch like this has literally 0 performance impact on this code and is absolutely fine, it might displace actually important branches out of the limited branch predictor cache.

For instance I have a PR open which simply removes `std::deque` and improves performance by ~5% on my CPU. This is really weird actually, because if you benchmark `std::deque` in a small benchmark project you'll find that it's super fast. And it's only iterated about ~100/sec! Still... it's mere presence has a negative impact, despite being fast itself. CPUs are dumb.

That said I wouldn't mind merging it as is. 😄